src/xcode/ENA/ENA/Source/Scenes/HealthCertificates/HealthCertifiedPerson/UpdateConsent/HealthCertifiedPersonUpdateConsentViewController.swift

```diff
@@ -0,0 +1,105 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+
+class HealthCertifiedPersonUpdateConsentViewController: UIViewController, DismissHandling, FooterViewHandling {
+
+	// MARK: - Init
+
+	init(
+		presentAlert: @escaping (_ ok: UIAlertAction, _ retry: UIAlertAction) -> Void,
+		presentUpdateSuccess: @escaping () -> Void,
+		didCancel: @escaping () -> Void,
+		dismiss: @escaping () -> Void
+	) {
+		self.presentAlert = presentAlert
+		self.viewModel = HealthCertifiedPersonUpdateConsentViewModel()
+		self.presentUpdateSuccess = presentUpdateSuccess
+		self.didCancel = didCancel
+		self.dismiss = dismiss
+		super.init(nibName: nil, bundle: nil)
+	}
+
+	@available(*, unavailable)
+	required init?(coder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	// MARK: - Overrides
+
+	override func viewDidLoad() {
+		super.viewDidLoad()
+		// Do any additional setup after loading the view.
+		navigationItem.hidesBackButton = true
+		navigationItem.rightBarButtonItem = dismissHandlingCloseBarButton
+		title = AppStrings.HealthCertificate.Person.UpdateConsent.title
+	}
+
+	// MARK: - Protocol DismissHandling
+
+	func wasAttemptedToBeDismissed() {
+		dismiss()
+	}
+
+	// MARK: - Protocol FooterViewHandling
+
+	func didTapFooterViewButton(_ type: FooterViewModel.ButtonType) {
+		switch type {
+		case .primary:
+			footerView?.setLoadingIndicator(true, disable: true, button: .primary)
+			viewModel.submit(
+				completion: { [weak self] result in
+					switch result {
+					case .success:
+						DispatchQueue.main.async {
+							self?.presentUpdateSuccess()
+						}
+					case .failure:
+						DispatchQueue.main.async {
+							self?.showAlert()
+						}
+					}
+				}
+			)
+		case .secondary:
+			dismiss()
+		}
+	}
+
+	// MARK: - Public
+
+	// MARK: - Internal
+
+	// MARK: - Private
+
+	private let presentAlert: (_ ok: UIAlertAction, _ retry: UIAlertAction) -> Void
```

Comment:
Perhaps the Alert should expect an error (I am defining in the resource) and then take the error's `localizedDescription` as the message? Like we do it in the `HealthCertificateCoordinator`.
Idea behind is that I define a specific DCCReissuanceResourceError with localized texts which we can pass through until here if the loading fails.

Reply:
See above